### PR TITLE
Fix bug #300: normalized data looks flat

### PR DESCRIPTION
**Pull request by riecke**
Summary of changes:
1. Add "min_value" and "max_value" to conditions configuration.
2. Add a function (get_min_max_values) in PillarConfig object.
3. Add PillarConfig to the conditions/views.py file, and look up the min/max values when processing a wms (get tile) request.
4. Pass the min/max values to the SQL function on the backend, and use them to rescale the values in the stored raster to between 0-254 (reserving 255 for "no data")

Also: fix typing problems in "conditions_test.py"

### CODE REVIEW

File: src/planscape/config/conditions_config.py [hunk 1]
```diff
@@ -17,8 +17,58 @@
...
+                   'display'}.union(COMMON_METADATA)
+    ELEMENT_KEYS = {'element_name',
+                    'metrics',
+                    'operation'}.union(COMMON_METADATA)
+    METRIC_KEYS = {'metric_name',
+                   'current_conditions_only',
+                   'ignore',
+                   'source',
+                   'source_link',
+                   'data_source',
```
Comment (elsiel23) [inline src/planscape/config/conditions_config.py:41]: if this is always a set of people or an organization, would data_owner be more descriptive?

File: src/planscape/base/condition_types.py [hunk 2]
```diff
@@ -50,6 +50,8 @@ class Metric(TypedDict):
     display_name: NotRequired[str]
     current_conditions_only: NotRequired[boolean]
     colormap: NotRequired[str]
+    min_value: NotRequired[int]
```
Comment (elsiel23) [inline src/planscape/base/condition_types.py:53]: should these classes/structs also be updated with the data source related fields? and should min/max value be added in Element/Pillar as well?

File: src/planscape/conditions/migrations/0002_get_rast_tile.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
+from django.db import migrations
```
Comment (elsiel23) [inline src/planscape/conditions/migrations/0002_get_rast_tile.py:1]: on my machine, this file is named conditions/migrations/get_rast_tile.py so the diff isn't showing up properly here. is that change intentional?

Comment (riecke) [reply src/planscape/conditions/migrations/0002_get_rast_tile.py:1]: Unfortunately!  Because it's a new version of the function, I had to create a new file for upload (this one is 0002_...). Let's see if I can get the diff to show up.

File: src/planscape/config/conditions_config.py (hunk 1 above)
Comment (riecke) [reply src/planscape/config/conditions_config.py:41]: I agree---"data_source" is too vague.  How about "data_provider"?

File: src/planscape/base/condition_types.py (hunk 2 above)
Comment (riecke) [reply src/planscape/base/condition_types.py:53]: Min/max aren't needed for elements and pillars, since they are always -1/1.

I don't think the data source fields are needed in the backend, just the frontend.  Can I add those in a separate PR if they're needed?

File: src/planscape/conditions/migrations/0002_get_rast_tile.py (hunk 3 above)
Comment (riecke) [reply src/planscape/conditions/migrations/0002_get_rast_tile.py:1]: I'm sorry!  I tried to fix the branch so that it shows only the diffs with the original file, but I couldn't get GitHub to see it.  Here is the diff:

https://github.com/OurPlanscape/Planscape/commit/8982eaea823b2c38108ad5f41fd367c1689599f5

Comment (elsiel23) [reply src/planscape/conditions/migrations/0002_get_rast_tile.py:1]: thank you for getting that to work!

File: src/planscape/config/conditions_config.py (hunk 1 above)
Comment (elsiel23) [reply src/planscape/config/conditions_config.py:41]: SG

File: src/planscape/base/condition_types.py (hunk 2 above)
Comment (elsiel23) [reply src/planscape/base/condition_types.py:53]: sg, let's wait and see

**Review by elsiel23 — APPROVED** [review]
